### PR TITLE
Breakout add blog to it's own component

### PR DESCRIPTION
**Pull request by tommyemr**
What I have below works, but interested in knowing if there is a better way to refresh the Blog list in the main component from the new AddBlog component.

### CODE REVIEW

File: FoxHound.Client/src/AddBlog.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    padding: theme.spacing(2, 3),
+  },
+  templateBuilder: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
```
Comment (jjh121) [inline FoxHound.Client/src/AddBlog.tsx:23]: Remove this as it looks like it isn't being used anywhere.

Comment (tommyemr) [reply FoxHound.Client/src/AddBlog.tsx]: Removed, thanks

**Comment by jjh121** [comment]
> What I have below works, but interested in knowing if there is a better way to refresh the Blog list in the main component from the new AddBlog component.

I think this works (maybe some others can chime in with other opinions). However, I may name this slightly differently. Consider something like "handleBlogAdded" or "blogAdded" in case we want to do more than just refresh the blog. You also get the blogId back, it would probably be beneficial to pass that to that method. I'm thinking long term we probably don't want to refresh the blog, but rather navigate to the newly created blog's page.

File: FoxHound.Client/src/AddBlog.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect, ChangeEvent, FormEvent } from "react";
+import axios from "axios";
+import { makeStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import Grid from "@material-ui/core/Grid";
+import Paper from "@material-ui/core/Paper";
+import Container from "@material-ui/core/Container";
+import Box from "@material-ui/core/Box";
+import Typography from "@material-ui/core/Typography";
```
Comment (jjh121) [inline FoxHound.Client/src/AddBlog.tsx:10]: Let's try and do import {x,  y, z} from @material-ui/core and not list them all out as separate imports. Ryan had mentioned this has gotten better for compilation times. Have you noticed any significant performance differences one way or another?

File: FoxHound.Client/src/AddBlog.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+}));
+
+const AddBlog: React.FC<IProps> = (props) => {
+  const [owner, setOwner] = useState<string>("");
+  const [title, setTitle] = useState<string>("");
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [error, setError] = useState<string>("");
+  const { refreshBlogList } = props;
+
+  useEffect(() => {}, []);
```
Comment (jjh121) [inline FoxHound.Client/src/AddBlog.tsx:29]: Can be removed

File: FoxHound.Client/src/AddBlog.tsx (hunk 2 above)
Comment (tommyemr) [reply FoxHound.Client/src/AddBlog.tsx:10]: I can make these change, I didn't notice any performance differences.

File: FoxHound.Client/src/AddBlog.tsx (hunk 3 above)
Comment (tommyemr) [reply FoxHound.Client/src/AddBlog.tsx:29]: 1. I am using setError within the try/catch for Adding the new blog.
2. 'refreshBlogList' constant could be removed and referenced below as props.refreshBlogList but I did it this way for consistency.
3. useEffect was not used, I will remove.

Comment (jjh121) [reply FoxHound.Client/src/AddBlog.tsx:29]: Sorry, i just tried highlighting useEffect...still getting used to this PR experience.

Comment (tommyemr) [reply FoxHound.Client/src/AddBlog.tsx]: no problem, just wanted to make sure I didn't miss a change.

**Pull request merged**